### PR TITLE
Create a new tweet

### PR DESCRIPTION
**Pull request by anastasiialisikh**

### CODE REVIEW

**Review by stanimirtt — APPROVED** [review]

File: src/components/CreateTweet/CreateTweet.tsx [hunk 1]
```diff
@@ -2,14 +2,38 @@ import * as React from 'react';
...
+interface ITweetProps {
+  stores?: IStores
+}
+
+@inject('stores')
+@observer
+export class CreateTweet extends React.Component<ITweetProps> {
+  constructor(props: ITweetProps) {
+    super(props);
+    this.handleCreateTweet = this.handleCreateTweet.bind(this);
```
Comment (fstodulski) [inline src/components/CreateTweet/CreateTweet.tsx]: You don't have to use
bind(this)

File: src/components/CreateTweet/CreateTweet.tsx [hunk 2]
```diff
@@ -2,14 +2,38 @@ import * as React from 'react';
...
+  }
+
+  handleCreateTweet() {
```
Comment (fstodulski) [inline src/components/CreateTweet/CreateTweet.tsx]: try 
`handleCreateTweet = () => {....}`

File: src/components/TweetItem/TweetItem.tsx [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
   tweet: ITweet,
+  stores?: IStores
 }
 
+@inject("stores")
+@observer
 export class TweetItem extends React.Component<ITweetItemProps, {}> {
   constructor(props: ITweetItemProps) {
     super(props);
+    this.handleRemoveTweet = this.handleRemoveTweet.bind(this);
```
Comment (fstodulski) [inline src/components/TweetItem/TweetItem.tsx]: the same, 
in declaring methods use
`methodName = () => {......}` instead - so you won't be needed to use`.bind(this)` in controller

File: src/stores/Tweets.ts [hunk 4]
```diff
@@ -1,9 +1,59 @@
...
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          'authorId': authorId,
+          'content': content
+        })
+      }
+    )
+    .then(() => {
```
Comment (fstodulski) [inline src/stores/Tweets.ts:35]: You don't need to fetch after posting. From server as response you will recive all Tweets including added one

File: src/stores/Tweets.ts [hunk 5]
```diff
@@ -1,9 +1,59 @@
...
+      this.fetchTweets();
+    });
+  }
+
+  @action
+  removeTweet(tweetId: number) {
+    return Fetch.request(
```
Comment (fstodulski) [inline src/stores/Tweets.ts:42]: The same. After deleting you will recive all Tweets from database as response

**Review by fstodulski — CHANGES_REQUESTED** [review]
Please check my comments.

File: src/stores/Tweets.ts (hunk 5 above)
Comment (anastasiialisikh) [reply src/stores/Tweets.ts:42]: It's not ok to receive the list on the response of request creating a new item, usually in this case response should be with 201 code and/or with an empty body or only with just created item.

File: src/stores/Tweets.ts (hunk 4 above)
Comment (anastasiialisikh) [reply src/stores/Tweets.ts:35]: the same

**Review by suezzo — APPROVED** [review]

**Review by fstodulski — APPROVED** [review]

**Pull request merged**